### PR TITLE
Unset inherited backgrounds on Posts Lists.

### LINKED ISSUE #37388: Global Styles: Setting color for List block impacts Post Template block
### Description

This was found as part of the [eleventh call for testing for the FSE Outreach Program](https://make.wordpress.org/test/2021/11/08/fse-program-testing-call-11-site-editing-safari/#comment-2254):

>Next up I clicked through the Styles options and decided to change the background color of the lists block. This resulted in the entire background of the posts list to change the background color. I was expecting that only the actual lists blocks would change when adjusting this setting, instead the entire page background of the query block changed. There’s already an issue about adding a “wp-block-list” class to the lists block and I created a PR for this too.

This is the PR referenced: https://github.com/WordPress/gutenberg/pull/36676  cc @oandregal as a heads up. 

### Step-by-step reproduction instructions

1. Open site editor and ensure a Query Loop block is placed within.
2. Open Styles > Blocks > Lists > Colors. Set a background color.
3. Notice the background color of the post template block changes within the query loop.

### Screenshots, screen recording, code snippet

https://user-images.githubusercontent.com/26996883/146113713-8d23ab03-211e-4d1c-8a2e-711ac6ddf889.mov

### Environment info

- WordPress 5.9 beta 3
- TT2
- No Gutenberg. 

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

### PR DESCRIPTION
**Pull request by jasmussen**
## Description

Makes progress on #37388, alternative to #37528, and goes well with #37940. 

If you set the List block to have a specific background color in global styles, the ul block is targetted, making that color bleed into the Posts List and Query Loop blocks:

before

This PR unsets any inherited colors on the `ul` element used:

Screenshot 2022-01-13 at 09 50 00

## How has this been tested?
Change the background color of the List block in global styles. Then insert a Posts List or Query Loop block, they should still have a transparent background.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal). 
- [ ] I've updated related schemas if appropriate.

### CODE REVIEW

File: packages/block-library/src/post-template/style.scss [hunk 1]
```diff
@@ -11,6 +11,11 @@
 	list-style: none;
 	padding: 0;
 
+	// Unset background colors that can be inherited from Global Styles with extra specificity.
+	&.wp-block-post-template {
+		background: none;
```
Comment (ntsekouras) [inline packages/block-library/src/post-template/style.scss:16]: I guess in this approach we should also handle text and link color too.

**Review by ntsekouras — COMMENTED** [review]
Thanks for the PR Joen!

The root problem of this is a tough one and by also reading the issues/discussions about it, there are many angles to it.

I feel like I'm leaning towards the alternative from @oandregal to exclude `ul/ol` which are blocks, but I could see a short term solution like this one for core blocks that don't have `color` support(?), if we don't find a better alternative soon. That would mean to apply such treatment to all core blocks that fall into this category - there are more than this one and social icons.

File: packages/block-library/src/post-template/style.scss (hunk 1 above)
Comment (jasmussen) [reply packages/block-library/src/post-template/style.scss:16]: > exclude ul/ol which are blocks

If we could find some solid and low-specificity code that worked in the editor and frontend, I'd be fine with that too 👍 👍

But at the same time, I fundamentally don't think the inheritance is necessarily a big problem.

**Review by oandregal — APPROVED** [review]
Looking at this block, it's a container of blocks that acts as an intermediary between the query and the posts themselves. It can't be nested with any block other than query.

The only scenario where I see this could be problematic is when a theme tries to target this block specifically for color. For example, if it has a margin/padding and the theme wants to assign the color to this instead of targeting the query. I've looked into some themes (including the default ones that bundle with core) and none does this.

Given that we're blocked as to how to solve the overall issue and this is an uncontroversial fix that moves things forward, I'd think this is something we should merge. I'd still like someone with a lot more theme developer experience than I have to confirm this is fine for themes.

Also, probably extending this to any other inherited property supported by the list block is worth exploring.

**Comment by jasmussen** [comment]
Cool, I agree that it's a very trivial fix and mostly harmless. I have edited the description of this PR so that it shouldn't close #37388 when merged. Let's land this and see how things feel!

**Comment by Mamaduka** [comment]
Backported to the minor.

**Pull request merged**